pkg/cloudprovider/aws/instance.go

```diff
@@ -105,6 +128,37 @@ func (p *InstanceProvider) Terminate(ctx context.Context, node *v1.Node) error {
 	return nil
 }
 
+// The intention is to remove Offerings from the provided possible InstanceTypes based on recently observed ICEs, which will
```

Comment:
same here w/ go docs style, start with func name. 

```
// DiscardICEdInstanceTypes
```

Also, you could probably shorten to `DiscardICEd(ctx, instanceTypes)` since instanceTypes will be passed in.  The ICEd is awkward looking, so maybe we could do `DiscardInsufficientCapacityFor(ctx, instanceTypes)` ?  I'm not sure, naming is hard :p